sardana_adlink/ctrl/AdlinkAIOneDCtrl.py

```diff
@@ -0,0 +1,440 @@
+                    self._synchronization == AcqSynch.HardwareTrigger:
+                self._log.warning('The Adlink finished but the ctrl did not '
+                                  'read all the data yet. Last index readed %r'
+                                  % self._last_index_read)
+                self._state = State.Moving
+                self._status = 'The Adlink is acquiring'
+            # else:
+            self._state = State.On
+            self._status = 'The Adlink is ready to acquire'
+        else:
+            self._state = from_tango_state_to_state(self._hw_state)
+            self._status = 'The Adlink state is haha: %s' % self._hw_state
+
+    @debug_it
+    def StateOne(self, axis):
+        return self._state, self._status
+
+    @debug_it
+    @handle_error(msg="StateAll: Unable to configure the device!")
+    def LoadOne(self, axis, value, repetitions, latency):
+        self._stop_device()
+        self._clean_acquisition()
+
+        self.intTime = value
+        self._repetitions = repetitions
+
+        sample_rate = self.AIDevice['SampleRate'].value
+        chn_samp_per_trigger = int(self.intTime * sample_rate)
+
+        if self._synchronization == AcqSynch.SoftwareTrigger:
+            if value <= self._start_wait_time:
+                msg = 'It is not possible to integrate less than %r in ' \
+                      'software synchronization' % self._start_wait_time
+                raise ValueError(msg)
+            source = "SOFT"
+        elif self._synchronization == AcqSynch.HardwareTrigger:
+            source = "ExtD:+"
+            if self._pointsperstep > 1:
+                self._repetitions = self._pointsperstep
+        else:
+            raise ValueError("Adlink daq2005 allows only Software or "
+                             "Hardware triggering")
+
+        self.AIDevice["TriggerInfinite"] = 0
+        self.AIDevice["TriggerSources"] = source
+        self.AIDevice["NumOfTriggers"] = self._repetitions
+        self.AIDevice['ChannelSamplesPerTrigger'] = chn_samp_per_trigger
+
+    @debug_it
+    def PreStartOne(self, axis, value=None):
+        if axis != 1:
+            self._master_channel = axis
+        return True
+    
+    @debug_it
+    @handle_error(msg="StartAll: Unable to start acquisition on the device!")
+    def StartAll(self):
+        """
+        Starting the acquisition is done only if before was called
+        PreStartOne for master channel.
+        """
+
+        if self._synchronization == AcqSynch.HardwareTrigger:
+            attr_name = 'C0%s_MeanValues' % (self._master_channel - 2)
+            cb = ListenerDataReady(self._index_queue, log=self._log)
+            event_type = tango.EventType.DATA_READY_EVENT
+            self._id_callback = self.AIDevice.subscribe_event(attr_name,
+                                                              event_type, cb)
+
+        # AdlinkAI Tango device has two aleatory bugs:
```

Comment:
Will this break if the bugs in the DS were fixed?

Reply:
good question, we are using the DS from Alba as well, i hope when the bugs are fixed the controller is updated as well :)